sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/FeatureManagementConfiguration.java

```diff
@@ -27,7 +34,19 @@ class FeatureManagementConfiguration {
      */
     @Bean
     FeatureManager featureManager(ApplicationContext context,
-        FeatureManagementProperties featureManagementConfigurations, FeatureManagementConfigProperties properties) {
-        return new FeatureManager(context, featureManagementConfigurations, properties);
+        FeatureManagementProperties featureManagementConfigurations, FeatureManagementConfigProperties properties,
+        ObjectProvider<VariantProperties> propertiesProvider) {
+
+        TargetingContextAccessor contextAccessor = appContext.getBeanProvider(TargetingContextAccessor.class)
+            .getIfAvailable();
+        TargetingEvaluationOptions evaluationOptions = appContext.getBeanProvider(TargetingEvaluationOptions.class)
+            .getIfAvailable();
```

Comment:
Why use getBeanProvider explicitly, instead of defining the ObjectProvider on the method's signature?